src/ddsrt/src/process/windows/process.c

```diff
@@ -0,0 +1,374 @@
+
+dds_retcode_t
+ddsrt_process_create(
+  const char *executable,
+  char *const argv[],
+  ddsrt_pid_t *pid)
+{
+  dds_retcode_t rv = DDS_RETCODE_ERROR;
+  PROCESS_INFORMATION process_info;
+  STARTUPINFO si;
+  char *cmd;
+  LPTCH environment;
+
+  assert(executable != NULL);
+  assert(pid != NULL);
+
+  cmd = commandline(executable, argv);
+
+  memset(&si, 0, sizeof(STARTUPINFO));
+  si.cb = sizeof(STARTUPINFO);
+
+  /* The new process will inherit the input/output handles. */
+  /* TODO: Redirect is not working yet. */
+  si.dwFlags = STARTF_USESTDHANDLES;
+  si.hStdOutput = GetStdHandle(STD_OUTPUT_HANDLE);
+  si.hStdInput  = GetStdHandle(STD_INPUT_HANDLE);
+  si.hStdError  = GetStdHandle(STD_ERROR_HANDLE);
+
+  /* Get the environment variables to pass along. */
+  environment = GetEnvironmentStrings();
+  if(environment){
+    BOOL created;
+    created = CreateProcess(executable,               // ApplicationName
+                            cmd,                      // CommandLine
+                            NULL,                     // ProcessAttributes
+                            NULL,                     // ThreadAttributes
+                            TRUE,                     // InheritHandles
+                            CREATE_NO_WINDOW,         // dwCreationFlags
+                            (LPVOID)environment,      // Environment
+                            NULL,                     // CurrentDirectory
+                            &si,                      // StartupInfo
+                            &process_info);           // ProcessInformation
+    if (created) {
+      *pid = process_info.dwProcessId;
+      rv = DDS_RETCODE_OK;
+    } else {
+      DWORD error = GetLastError();
+      if ((ERROR_FILE_NOT_FOUND == error) ||
+          (ERROR_PATH_NOT_FOUND == error) ||
+          (ERROR_ACCESS_DENIED  == error)) {
+        rv = DDS_RETCODE_BAD_PARAMETER;
+      }
+    }
+    FreeEnvironmentStrings(environment);
+  }
+
+  ddsrt_free(cmd);
+
+  return rv;
+}
+
+
+
+dds_retcode_t
+ddsrt_process_wait_exit(
+  ddsrt_pid_t pid,
+  dds_duration_t timeout,
+  int32_t *status)
+{
+  dds_retcode_t rv = DDS_RETCODE_BAD_PARAMETER;
+  HANDLE phdl;
+
+  phdl = pid_to_phdl(pid);
+  if (phdl != 0) {
+    rv = process_wait_exit(phdl, timeout, status);
+    CloseHandle (phdl);
+  }
+
+  return rv;
+}
+
+
+
+dds_retcode_t
+ddsrt_process_terminate(
+  ddsrt_pid_t pid,
+  dds_duration_t timeout)
+{
+  dds_retcode_t rv = DDS_RETCODE_BAD_PARAMETER;
+  HANDLE phdl;
+
+  phdl = pid_to_phdl(pid);
+  if (phdl != 0) {
+    /* Try a graceful kill. */
+    rv = process_terminate(phdl, timeout);
+
+    /* Escalate when not killed yet. */
+    if (rv != DDS_RETCODE_OK) {
+      /* Forcefully kill. */
+      rv = process_kill(phdl, timeout);
+      if (rv == DDS_RETCODE_OK) {
+        rv = DDS_RETCODE_TIMEOUT;
+      }
+    }
+    CloseHandle(phdl);
+  }
+
+  return rv;
+}
+
+
+
+#if (_WIN32_WINNT <= 0x0502)
```

Comment:
I am not interested in XP support, but I probably would accept a PR that adds XP support throughout if someone had a need for it and contributed it.

Reply:
So, I'll keep it for now...

Reply:
We shouldn't. If someone comes along and wants to add support for a legacy operating system, that's fine. But we dropped XP support in a lot of other places already, so keeping it would only cause confusion. Windows 7 will reach EOL in less than a year...